src/pages/Profile.js

```diff
@@ -1,22 +1,57 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, {useContext, useEffect, useState} from 'react';
+import {Link} from 'react-router-dom';
+import {AuthContext} from "../context/AuthContext";
+import axios from "axios";
 
 function Profile() {
-  return (
-    <>
-      <h1>Profielpagina</h1>
-      <section>
-        <h2>Gegevens</h2>
-        <p><strong>Gebruikersnaam:</strong> hardcoded-test</p>
-        <p><strong>Email:</strong> hardcoded@test.com</p>
-      </section>
-      <section>
-        <h2>Strikt geheime profiel-content</h2>
-        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
-      </section>
-      <p>Terug naar de <Link to="/">Homepagina</Link></p>
-    </>
-  );
+
+    const [privateContent, setPrivateContent] = useState({})
```

Comment:
Ik zou de default waarde `undefined` maken, dan heb je een beter verschil tussen of het al wel of niet is ingeladen.

Op dit moment krijg je ook errors bij de return als het request niet lukt, en hij `username` en `email` niet kan vinden op het object.

```suggestion
    const [privateContent, setPrivateContent] = useState();
```